include/boost/url/param.hpp

```diff
@@ -742,96 +690,93 @@ struct param_pct_view
 
         @par Example
         @code
-        param_pct_view qp( "key" );
+        param_pct_view qp( "key", "value" );
         @endcode
 
         @par Postconditions
         @code
-        this->key.data() == key.data() && this->value == "" && this->has_value == false
+        this->key.data() == key.data() && this->value.data() == value.data() && this->has_value == true
         @endcode
 
         @par Complexity
-        Linear in `key.size()`.
+        Linear in `key.size() + value.size()`.
 
         @par Exception Safety
         Exceptions thrown on invalid input.
 
         @throw system_error
-        `key` contains an invalid percent-encoding.
-
-        @param key The key to set.
+        `key` or `value` contains an invalid percent-encoding.
 
-        @see
-            @ref no_value.
+        @param key, value The key and value to set.
     */
-    /**@{*/
-    param_pct_view(
-        pct_string_view key,
-        no_value_t) noexcept
-        : key(key)
-    {
-    }
-
     param_pct_view(
         pct_string_view key,
-        std::nullptr_t) noexcept
+        pct_string_view value) noexcept
         : key(key)
+        , value(value)
+        , has_value(true)
     {
     }
-    /**@}*/
 
     /** Constructor
 
         This constructs a parameter with a key
-        and value, which may both contain percent
-        escapes.
+        and optional value, which may both
+        contain percent escapes.
+
         The new key and value will reference
         the same corresponding underlying
         character buffers.
+
         Ownership of the buffers is not transferred;
         the caller is responsible for ensuring that
         the assigned buffers remain valid until
         they are no longer referenced.
 
         @par Example
         @code
-        param_pct_view qp( "key", "value" );
+        param_pct_view qp( "key", optional<string_view>("value") );
         @endcode
 
         @par Postconditions
         @code
-        this->key.data() == key.data() && this->value.data() == value.data() && this->has_value == true
+        this->key.data() == key.data() && this->value->data() == value->data() && this->has_value == true
         @endcode
 
         @par Complexity
-        Linear in `key.size() + value.size()`.
+        Linear in `key.size() + value->size()`.
 
         @par Exception Safety
         Exceptions thrown on invalid input.
 
         @throw system_error
         `key` or `value` contains an invalid percent-encoding.
 
+        @tparam OptionalLike An optional
+        @ref string_view type, such as
+        `boost::optional<string_view>` or
+        `std::optional<string_view>`.
+
         @param key, value The key and value to set.
     */
+    template <class OptionalLike>
     param_pct_view(
         pct_string_view key,
-        pct_string_view value) noexcept
-        : key(key)
-        , value(value)
-        , has_value(true)
+        OptionalLike const& value)
```

Comment:
I'm not sure I like seeing `OptionalLike`. What about `OptionalString`?

Reply:
Yes. `OptionalLike` was actually a mistake. I changed all others to `OptionalString` but forgot about this one.